dora/core/client/fs/src/main/java/alluxio/client/file/FileSystem.java

```diff
@@ -179,9 +177,10 @@ public static FileSystem create(FileSystemContext context) {
     public static FileSystem create(FileSystemContext context, FileSystemOptions options) {
       AlluxioConfiguration conf = context.getClusterConf();
       checkSortConf(conf);
-      FileSystem fs = options.getUfsFileSystemOptions().isPresent()
-          ? new UfsBaseFileSystem(context, options.getUfsFileSystemOptions().get())
-          : new BaseFileSystem(context);
+      Optional<UfsFileSystemOptions> ufsOptions = options.getUfsFileSystemOptions();
+      Preconditions.checkArgument(ufsOptions.isPresent(),
+          "Missing UfsFileSystemOptions in FileSystemOptions");
+      FileSystem fs = new UfsBaseFileSystem(context, options.getUfsFileSystemOptions().get());
```

Comment:
We should probably move these implicit dependencies on certain property keys to a start-up property check. Some other keys need to be consolidated together, so this matter belongs to a separate follow-up PR.

Reply:
I will change this Optional to a required non-null field, in another PR